dd-sdk-android-internal/src/main/java/com/datadog/android/internal/telemetry/InternalTelemetryEvent.kt

```diff
@@ -56,4 +59,21 @@ sealed class InternalTelemetryEvent {
     }
 
     object InterceptorInstantiated : InternalTelemetryEvent()
+
+    companion object {
+
+        /* The sampling rate is used when creating metrics.
+         * Creating(head) sampling rate exist only for long-lived metrics like method performance.
+         * Creating(head) sampling rate is responsible for metric creation.
+         * Created metric still could not be sent.
+         * Sending depends on [REPORTING_SAMPLING_RATE_KEY] sample
+         */
+        const val CREATION_SAMPLING_RATE_KEY: String = "HEAD_SAMPLING_RATE_KEY"
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏ I really do not understand the need of this sampling rate. I think I will need to read more about it, let's talk in private.

Reply:
Adusted doc a bit.

Overall, the reason why we need two sample rate is: 

> We have some metrics which sampling happens not from the place we call `logMetric`  but when we create such metric. For instance - `MethodCalledTelemetry`. It will be sampled at `SdkInternalLogger.startPerformanceMeasure()` but sent with `MethodCalledTelemetry.stopAndSend`. So the sampling rate for it will equal to `samplingRate` of `startPerformanceMeasure` method. 

Nevertheless there is a possibility to still sample metric at the `MethodCalledTelemetry.stopAndSend` so to compute effectiveSampleRate we need both of this values composed together